### PR TITLE
Switch to ruff for linting

### PR DESCRIPTION
**Pull request by jchristgit**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,9 +16,9 @@ repos:
       - id: python-check-blanket-noqa
```
Comment (ChrisLovering) [inline .pre-commit-config.yaml]: Cant comment higher, but pygrep hook can be removed from pre-commit as ruff covers it already (rule PGH)

**Review by ChrisLovering — DISMISSED** [review]
Looks good, just one optional thing that might be good to do now too.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jchristgit) [reply .pre-commit-config.yaml]: Thanks! I dropped it in the amended commit. The tests have also been fixed.

**Review by ChrisLovering — APPROVED** [review]

**Comment by jchristgit** [comment]
MASCHINNNNNNNNNNNNNNNN

**Pull request merged**